datafusion/physical-expr/src/aggregate/median.rs

```diff
@@ -111,13 +112,28 @@ impl PartialEq<dyn Any> for Median {
 /// The intermediate state is represented as a List of those scalars
```

Comment:
Can you please update the comments to reflect the change here (aka that the state is is stored as ScalarValues, but the conversion is delayed if possible)